cmds/report.c

```diff
@@ -20,6 +21,227 @@ enum {
 /* maximum length of symbol */
 static int maxlen = 20;
 
+static LIST_HEAD(output_fields);
+
+static void print_total_time(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->total.sum;
+
+	print_time_unit(d);
+}
+
+static void print_total_time_avg(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->total.avg;
+
+	print_time_unit(d);
+}
+
+static void print_total_time_min(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->total.min;
+
+	print_time_unit(d);
+}
+
+static void print_total_time_max(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->total.max;
+
+	print_time_unit(d);
+}
+
+static void print_self_time(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->self.sum;
+
+	print_time_unit(d);
+}
+
+static void print_self_time_avg(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->self.avg;
+
+	print_time_unit(d);
+}
+
+static void print_self_time_min(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->self.min;
+
+	print_time_unit(d);
+}
+
+static void print_self_time_max(struct field_data *fd)
+{
+	struct uftrace_report_node *node = fd->arg;
+	uint64_t d;
+
+	d = node->self.max;
+
+	print_time_unit(d);
+}
```

Comment:
Please consider adding a macro to define these functions - they are all same except the field name.  Please check `SORT_KEY` macro in utils/report.c file.

Reply:
Do you agree to define different macros for the base case and --diff case?

Reply:
Yes, I do.